### DISCUSSION TITLE
Welcome to tsun-gen3-proxy Discussions! (Write in German or English, pls)

### DISCUSSION THREAD

**Discussion by s-allius** [s-allius/tsun-gen3-proxy#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mrtiago** [comment]
hi, how can I generate and share the trace with you guys? I have 2 ms-2000

**Reply by s-allius** [reply]
Hello mrtiago,

unfortunately the proxy doesn't support the ms-2000 inverters by now.
I also have a ms-2000 in operation and have made the needed traces. I think, that I can add the ms-2000 support, but I need some time for it. At the moment there are some important projects that I have to finish before.
So my plan is to go on with this in April this year.

Cheers Stefan

**Comment by mrtiago** [comment]
Hello Stefan, that’s great

If you need anything that I can help with let me know

Tiago Rodrigues

**Comment by dreisieben** [comment]
Hi Stefan, thanks very much for this great project. I'm thinking about buying some TITAN TSOL-MP3000 units. Do you know if they could be supported as well and how much work would be involved? I'm mainly asking about the feasibility to help me decide whether to buy them or go with another solution. I might be able to help with the implementation (if needed or desired). Thanks.

**Reply by s-allius** [reply]
Hi dreisieben,
I never saw a MP-3000 or a trace of it. I think that this Inverter is very similar, to the new 3+ Generation (like the MS-2000), cause I saw both devices in the description of the new Talent App. But I didn't know it!
For the MS-2000 I have analysed the protocol and I think that I can implement the protocol. But I can't start with the implementation shortly and I'm not sure which data we will get.

When you can get a Wireshark trace of an MP-3000 I can have a look on it.

**Comment by dreisieben** [comment]
Hi Stefan, thank you. As soon as I get an MP3000 (which will take some time) I'll report back here.

**Comment by bernif68** [comment]
Hallo Stefan, ich hab einen TSOL-MS800 erworben und ihn versucht in HA einzubinden. 
Meine Seriennummer beginnt mit T47 uns somit kann ich die Konfig nicht speichern.
Fehler beim Speichern der Add-on Konfiguration, does not match regular expression ^(R17|Y17|Y47).{13}$. Got {'inverters': [{'serial': 'T47....

Der Datenlogger beginnt mit R47

Beste Grüße Bernhard

**Reply by s-allius** [reply]
Hallo Bernhard,

die R47 Seriennummer hinzufügen ist kein Problem. Wir müssen dann aber das Mapping der Register genau prüfen, dazu wäre ein Trace wichtig, den du mir bereitstellen müsstest...
Für einen ersten Test kannst du die Seriennummerprüfung ausschalten. Dazu einfach den Parameter `inverters.allow_all = true`in der config.yaml setzen. Wenn du das Add-on nutzt, findest du den Schalter `Erlaube Verbindungen von sämtlichen Wechselrichtern` in der Konfiguration. Bitte dann diese einschalten.

Wenn du dein Netzwerkkonfiguration (DNS) stimmt, solltest du in der proxy Konsole schnell Einsträge über Verbindungen von deinem WR sehen. 
Dann nochmal melden und wir sprechen über die nächsten Schritte.

**Reply by bernif68** [reply]
Hallo Stefan, danke für die rasche Antwort.  Ich nutze das Addon.
Netzwerk Konfig sollte passen. 

AdGuard:
13:28:09
1.3.2025
logger.talent-monitoring.com
Typ: A, Einfaches DNS
Umgeschrieben
0 ms
[192.168.1.101](http://homeassistant.local:8123/api/hassio_ingress/8RpNw1C6T1OB3JCC5OSF2ZQfr0jvcnpfj2AuFF_Zgyo/#logs?search=%22192.168.1.101%22)


`Add-on: TSUN-Proxy
 MQTT Proxy for TSUN Photovoltaic Inverters
-----------------------------------------------------------
 Add-on version: 0.12.1-rel
 You are running the latest version of this add-on.
 System: Home Assistant OS 14.2  (amd64 / qemux86-64)
 Home Assistant Core: 2025.2.5
 Home Assistant Supervisor: 2025.02.4
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
Add-on environment started
check for Home Assistant MQTT
[17:05:05] ERROR: Got unexpected response from the API: Service not enabled
MQTT not found
Start Proxyserver...
'2025-02-28 17:05:07  INFO | root | Server "proxy - 0.12.1-rel" will be started'
'2025-02-28 17:05:07  INFO | root | current dir: /home/proxy'
'2025-02-28 17:05:07  INFO | root | config_path: /homeassistant/tsun-proxy/'
'2025-02-28 17:05:07  INFO | root | json_config: /data/options.json'
'2025-02-28 17:05:07  INFO | root | toml_config: None'
'2025-02-28 17:05:07  INFO | root | log_path:    /homeassistant/tsun-proxy/logs/'
'2025-02-28 17:05:07  INFO | root | log_backups: 2 days'
'2025-02-28 17:05:07  INFO | root | LOG_LVL    : INFO'
'2025-02-28 17:05:07  INFO | root | ******'
'2025-02-28 17:05:07  INFO | root | Read from default_config.toml => ok'
'2025-02-28 17:05:07  INFO | root | Read from environment => ok'
'2025-02-28 17:05:07  INFO | root | Read from /homeassistant/tsun-proxy/config.json => n/a'
'2025-02-28 17:05:07  INFO | root | Read from /homeassistant/tsun-proxy/config.toml => n/a'
'2025-02-28 17:05:07  INFO | root | Read from /data/options.json => ok'
'2025-02-28 17:05:07  INFO | root | ******'
'2025-02-28 17:05:07  INFO | root | listen on port: 5005 for inverters'
'2025-02-28 17:05:07  INFO | root | listen on port: 10000 for inverters'
'2025-02-28 17:05:07  INFO | mqtt | start MQTT: host:192.168.1.75  port:1883  user:mqtt-user'
'2025-02-28 17:05:07  INFO | root | HTTP server listen on port: 8127'
'2025-02-28 17:05:07  INFO | mqtt | MQTT broker connection established'
'2025-02-28 17:05:07  INFO | root | Initialize proxy device on home assistant'
'2025-03-01 00:00:00  INFO | root | Clear daily counters at midnight'
'2025-03-01 06:49:02  INFO | conn | [G3:0] Accept connection from ('192.168.1.101', 58501)'
'2025-03-01 06:49:24  INFO | conn | [G3:1] Accept connection from ('192.168.1.101', 53607)'`

**Reply by bernif68** [reply]
Hallo Stefan, soweit ich das jetzt sehe funktioniert dein Proxy auch mit R47 einwandfrei! 
Ich habe Controller, Micro Inverter, Module PV1, Module PV2 und den Proxy als Geräte in Mqtt.
Alle Sensoren liefern Werte. Wäre super wenn du die R47 hinzufügen könntest.

Beste Grüße und danke für die tolle Arbeit.
Bernhard

**Reply by s-allius** [reply]
Danke fürs Feedback. Ich werde die 'R47' Versionsnummern freischalten. Kommt dann mit der version 0.13 und ist in der letzte Dev version schon mit drin

**Comment by Carlo-Thomas** [comment]
Hallo Stefan,
erst einmal Lob an dich für das tolle Projekt und die Zeit & Energie die hier reingesteckt wird.

Bei mir läuft HomeAssistant auf einer Nas in virtueller Maschine und ich nutze das AdGuard AddOn. Aktuell sind ein TSUN MS800 (R17xxx) sowie eine DCU1000 (410xxx) über deinen Proxy in der 0.13 dev. Version eingeschleift. Funktioniert nach etwas rumprobieren mit den Einstellungen in HASS, AdGuard und dem Proxy ohne Probleme, letztendlich genau wie in der Beschreibung 👍. 

Die DCU braucht aktuell eine Verbindung nach Fernost um die Einspeiseleistung von "Hand" bzw. über Zeitfenster in der App einstellen zu können (Std. Abgabe 150W). Auch ist die Lade-/Einspeisestrategie der DCU nicht wirklich zufriedenstellend, bei voller Batterie wird die PV Leistung abgeregelt / abgeschaltet und es fließt nur die in der DCU eingestellte Leistung ins Hausnetz anstatt den "Überschuss" komplett weiterzugeben (Firmware LSW5_01_3026_NS_05_01.00.00.00).
 
Ist in deinem Projekt geplant eine Anpassung der DCU Einspeiseleistung vornehmen zu können? Ist das überhaupt realisierbar?
Ich denke mit meinem 3EM direkt hinter dem Stromzähler wäre eine zeitnahe Anpassung an den Verbrauch über alle 3-Phasen bzw. eine Bypass-Funktion bei voller Batterie mit HomeAssistant realisierbar. Alle notwendigen Daten um mit etwas "Automatisierung" an die eigenen Bedürfnisse anzupassen sind vorhanden (Stromverbrauch, Ladezustand Batterie, PV Leistung). Ziel wäre nicht eine 0 Einspeisung zu erreichen, dafür arbeiten Wechselrichter etc. sowieso zu langsam. 

Hast du eine DCU1000 vor Ort, kannst du Zuarbeit für die DCU gebrauchen? Ich bin aktuell zwar mit Wireshark nicht bewandert, das kann man aber ändern.

Gruß aus Norddeutschland

Carlo

**Reply by s-allius** [reply]
Hallo Carlo, schön, dass dir das Projekt gefällt.

Ich selbe habe keine DCU-1000, habe allerdings eine Weiterleitung von einem Anwender, sodass ich die Daten in meinem Proxy hier loggen und auswerten kann. Die Meisten Register habe ich schon rausgefunden. Allerdings vermisse ich noch:

- Status der Batterie Heizung (on/off)
- Temperaturen (Batterie, Außen, Elektronik)
- Bedeutung zweier Status-Register (Status-1 und Status-2 in der aktuellen 0.13-dev), welche die Werte 0,1 und 2 annehmen

Gefunden habe ich das Register 0x70, welches vermutlich den Status des Ausgangs anzeigt. Ich habe es heute in Out-Status ungenannt
Das Register 0x74 nimmt sehr sprunghaft Werte 0 und 2048 an. Dabei wird bei 2048 sofort der Ausgang deaktiviert. Ich habe aber auch einmal einen Wert von  dazwischen gesehen. Ich vermute das es den Eingangswert zur Regelung darstellt und direkt vovom Smart Meter kommt.
Die Temperaturen vermute ich auf den Registern 0x68,0x6a und 0x72:

![9C14EE1A-6B83-4C2D-BECC-9E59B8077869](https://github.com/user-attachments/assets/0ea73e41-4496-4677-87ae-ba0346429ecb)

Wenn du magst, kannst du gerne helfen die Bedeutung der register zu bestimmen (Dann sollten wir einen separaten Threat aufmachen)

Zur Regelung:
Ich weiß, dass der Zero Export vermutlich über eine separates Protokoll zwischen Smart Meter und DCU oder Inverter gesteuert wird. Die Kommunikation des Smart Meters ist aber verschlüsselt. Ich habe mir ein Smart Meter von TSUN bestellt, um dies genauer untersuchen zu können. Mit etwas Glück ist im Auslieferungszustand die Kommunikation noch nicht verschlüsselt. Bei meinem Inverter wurde die TLS Fw auch erst durch ein Update aufgespielt...

Gruß Stefan

**Reply by Carlo-Thomas** [reply]
Hallo Stefan,

kann dir aktuell ein paar HA Auszüge für die Register 74 => Status 1/2 sowie 68, 6a, 72 reinstellen.
Dein Projekt, dein Bereich, mach bitte einen Threat auf wo es für dich am besten passt, ich schieb die Infos dann rein.

Helfe im Rahmen meiner Möglichkeiten gern mit den Daten die meine DCU ausspuckt.

Gruß

Carlo

**Reply by s-allius** [reply]
Gerne, wir können hier: https://github.com/s-allius/tsun-gen3-proxy/discussions/310#discussion-8090193 Infos sammeln und diskutieren

**Comment by mrtool42** [comment]
Servus Stefan!

Erstmal Hut ab vor diesem coolen Proxy :D

Ich betreibe Homeassistant schon länger unter Docker und habe auch schon einen MQTT-Broker laufen.
Soweit habe ich auch eine Verbindung in beide Richtungen und die Werte kommen überall an.
Nur die Polling-Frequency stört mich noch ein bisschen - kann man die noch irgendwo anpassen?
Alle 5min ist mir einfach zu wenig.
Und könnte man theoretisch auch die Leistung des WR begrenzen? (TSOL-MS800-D)

Beste Grüße
Mario

**Reply by s-allius** [reply]
Hallo Mario,

du kannst du Leistung des Wechselrichters einstellen, indem du die `Rated Load` setzt: https://github.com/s-allius/tsun-gen3-proxy/wiki/MQTT-topics#inverter-rated-load 
Je nach Modell variieren etwas die Einstellungen, daher musst du da Werte mal durch probieren.

Die Abtastrate kannst du verkürzen, indem du in der Configuration den `modbus_polling` Mode aktivierst: https://github.com/s-allius/tsun-gen3-proxy/wiki/Configuration-toml#inverter-definitions

Gruß Stefan

**Comment by JazzMetal** [comment]
Hallo Stefan, tolles Projekt, vielen Dank für deine Arbeit!
Habe HA in einer VM laufen, und möchte einen MS-2000 ansteuern. 
Ich habe alles installiert, AdGuard lenkt die DNS Anfrage auf HA um, Connection zum MQTT scheint zu laufen, im Explorer Kann ich  tsun sehen. 
allerdings sehe ich im homeassistant keine Daten. 
habe jetzt m..E.. alles an Doku gelesen. Was kann ich jetzt wo nachschauen, um das Problem zu finden.
Vielen Dank, Werner

**Reply by s-allius** [reply]
Hallo Werner,

bei den MS-2000 wird vermutlich schon eine Firmware mit einer verschlüsselten Datenübertragung (TLS) installiert sein. In diesen Fall siehst du keinerlei reinkommende Verbindungen im Proxy Trace. Das liegt daran, dass ein anderer Port benutzt wird und der Proxy sich nicht dazwischen schalten kann. 
Dann kann der [Client Mode](https://github.com/s-allius/tsun-gen3-proxy/wiki/Operation-Modes-Overview#client-mode) helfen. Hierbei baut der Proxy  selbst eine Verbindung zum Proxy auf. Hast du diesen konfiguriert?

**Reply by JazzMetal** [reply]
Hallo Stefan,
danke fuer die schnelle Antwort. Nein - habe nicht den client mode konfiguriert.
ferner scheitere ich miserably daran, die IP Adresse meines Konverters zu identifizieren, denn die brauche ich dann wohl.
Schaue mir client mode config an und melde mich wieder,
vielen Dank
Werner

**Reply by JazzMetal** [reply]
Hallo Stefan,
vielen Dank, der client mode hat funktioniert, sobald ich die IP Adresse des converters hatte!
das war das eigentliche Problem - habe extrem viele Geräte und mit meiner einfachen Denke und selbstbestrickten tools bin ich nicht drangekommen.
ein LAN Mitschnittt in der FritzBox und Analyse in Wireshark hat dann geholfen
jetzt sehe ich im HA den converter und seine Daten
nochmals vielen Dank!!
Werner

**Reply by s-allius** [reply]
Schön, dass es nun läuft.

**Comment by JP83330** [comment]
Hello, I am quite a newbee with HA....so I have  TSOl MS2000 inverters and I had no problem to install Tsun_gen3 on my HA.

I can access to the dashboard but not input IP adress and so on....May I do a config file before ?

Thanks for a little help....

**Reply by s-allius** [reply]
Hi JP83330, with a MS2000 you must configure the Proxy to use the Client-Mode. Then the proxy establishes a connection to you inverter. For this your inverter needs a fixed IP address.

You find a short description here: https://github.com/s-allius/tsun-gen3-proxy/wiki/Operation-Modes-Overview#client-mode

**Comment by muelleraaron97-byte** [comment]
Hallo Stefan,

ich versuche aktuell den TSUN DCU2000Lite Speicher auch über den TSUN Proxy einzupflegen, stoße dabei jedoch auf Probleme. Ist das überhaupt möglich? Falls ja, welche Informationen brauchst du von mir?

**Reply by s-allius** [reply]
Hallo, wenn du das HA-AddOn nutzt, kannst du auf die aktuelle Dev Version 0.15 gehen. Da habe ich schon erste Änderungen für die DCU2000Lite eingebaut. Allerdings funktioniert das Auslesen der Werte noch nicht, da ich hier noch die Registerbelegen bestimmen muss. Wenn du magst kannst du mir da helfen, da ich selber keine DCU habe.

Lies einfach hier: https://github.com/s-allius/tsun-gen3-proxy/discussions/525

**Reply by aaronmue** [reply]
Hallo,

ja das kann ich gerne tun wenn du mir da behilflich bist gerne.

**Reply by s-allius** [reply]
Danke, dass du helfen willst. Ich schreibe dir mal im anderen Thread, wie du vorgehen kannst.

**Comment by JP83330** [comment]
Hello, I use TSUn GEN3 with TSOl MS2000 inverters, work very well but I have a question, is it possible to drive the MS2000 to restrict the output power ?  Why this because if you sell your electricity if the price is negative I don't want to pay for production !
Thanks for a little answer.

**Reply by s-allius** [reply]
Hi @JP83330 ,

you can set the output coefficient to reduce the output power: https://github.com/s-allius/tsun-gen3-proxy/wiki/MQTT-topics#inverter-output-coefficient

Attention: If you set this value, it will be stored in the Storage (Flash) of your inverter, like every other configuration setting. If you write very often into the flash you will damage your inverter in short time. So for a dynamic regulation, this Is the wrong approach. 

In this case you can get from TSUN a power meter for zero export. This might be the better solution for you.

Regards Stefan

**Reply by JP83330** [reply]
I understand that will be better to use smart meter tsun but as I am using HA to obtain negative electricity prices, any ideas if it is possible to drive the smart meter with HA to automatised the processus (prices are updated each 15 min wil be more simple than by hand)
Have a good day

**Comment by JP83330** [comment]
Many thanksBonne journee

Envoyé par AOL sur Android 
 
  Le mer., mai 27, 2026 à 23:16, Stefan ***@***.***> a écrit :   


Hi @JP83330 ,

you can set the output coefficient to reduce the output power: https://github.com/s-allius/tsun-gen3-proxy/wiki/MQTT-topics#inverter-output-coefficient

Attention: If you set this value, it will be stored in the Storage (Flash) of your inverter, like every other configuration setting. If you write very often into the flash you will damage your inverter in short time. So for a dynamic regulation, this Is the wrong approach.

In this case you can get from TSUN a power meter for zero export. This might be the better solution for you.

Regards Stefan

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!
